File: 알고리즘/버블정렬/BubbleSort.java
public class BubbleSort{
    public static void bubbleSort(int[] data){
        int n = data.length;
        for(int i=0; i<n-1; i++){
            boolean swapped = false;
            for(int j=0; j< n-i-1; j++){
                if(data[j] > data[j+1]){
                    int temp = data[j];
                    data[j] = data[j+1];
                    data[j+1] = temp;
                    swapped = true;
                }
            }
            if(!swapped){
                break;
            }
        }
    }

    public static void main(String[] args){
        int[] arr = {5,1,4,2,8};
        bubbleSort(arr);
        System.out.print("정렬 후 배열 : ");
        for(int num : arr){
            System.out.print(num + " ");
        }
    }
}

File: 알고리즘/버블정렬/bubbleSort.kt
fun bubbleSort(data : IntArray){
    val n = data.size
    for(i in 0 until n){
        var swapped = false
        for(j in 0 until n-i-1){
            if (data[j] > data[j+1]){
                val temp = data[j]
                data[j] = data[j+1]
                data[j+1] = temp
                swapped = true
            }
        }
        if (!swapped) break
    }
}

fun main(){
    val arr = IntArrayOf(5,1,4,2,8)
    println("정렬 전 배열 : ${arr.joinToString(", ")}")
    bubbleSort(arr)
    println("정렬 후 배열 : ${arr.joinToString(", ")}")
}